### PR TITLE
Fix #5281: Include ticks in tight bounding box

### LINKED ISSUE #5281: plt.tight_layout(pad=0) cuts away outer ticks
Wenn I do `plt.tight_layout(pad=0)` outer ticks are not considered and cut away:

```{python}
import matplotlib.pyplot as plt

fig, ax = plt.subplots()

ax.get_yaxis().set_tick_params(direction='out')
ax.get_xaxis().set_tick_params(direction='out')
ax.set_xlim(0.05, 0.95)

fig.tight_layout(pad=0)
fig.savefig('outer_ticks.png', dpi=300)
```

Result:

outer_ticks

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: lib/matplotlib/lines.py [hunk 1]
```diff
@@ -568,7 +568,7 @@ def get_window_extent(self, renderer):
                                  ignore=True)
         # correct for marker size, if any
         if self._marker:
-            ms = (self._markersize / 72.0 * self.figure.dpi) * 0.5
```
Comment (tacaswell) [inline lib/matplotlib/lines.py]: What is this change for?  I assume the issue is that with out this out ticks still get chopped?

**Comment by tacaswell** [comment]
Also breaks every single tight_layout test

File: lib/matplotlib/lines.py (hunk 1 above)
Comment (mdboom) [inline lib/matplotlib/lines.py]: Yes.  The assumption that markers only fill (-0.5, 0.5) is only true for circles.  Ticks fill (0, 1) or (-1, 0) depending on "in" or "out".

**Comment by mdboom** [comment]
Not surprised it breaks tight layout tests.  Will probably need to tie this change to `_internal.classic_mode`.

**Pull request closed without merging**